src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php

```diff
@@ -180,10 +184,16 @@ public function mapViolation(ConstraintViolation $violation, FormInterface $form
                     }
 
                     if (null !== $this->translator) {
+                        $form = $scope;
+
+                        do {
+                            $translationDomain = $form->getConfig()->getOption('translation_domain');
+                        } while (null === $translationDomain && null !== $form = $form->getParent());
```

Comment:
nit-pick:  The loop is a duplicate of the above one.  Introduce a private function?
```
private function getOption($form, $optionLabel) {
  do {
    $option = $form->getConfig()->getOption($optionLabel);
  } while (null === $option && null !== $form = $form->getParent());

  return $option;
}

```